File: iosApp/Offhand/UI/ProCrown.swift
import SwiftUI

// The gold crown that marks a Pro-only action for a free user; the same art
// ships as ic_pro_crown on Android. Gold is a brand mark, not a tinted glyph.
struct ProCrown: View {
    var size: CGFloat = 18

    var body: some View {
        Image("ProCrown")
            .renderingMode(.original)
            .resizable()
            .scaledToFit()
            .frame(width: size, height: size)
            .accessibilityLabel(String(localized: "Offhand Pro"))
    }
}

// The gating marker: a "PRO" pill placed right after a feature's title. The
// crown stays for the tier itself (paywall, Settings card).
struct ProBadge: View {
    var body: some View {
        Text(verbatim: "PRO")
            .font(.caption2.weight(.bold))
            .kerning(0.6)
            .foregroundStyle(Brand.proGold)
            .padding(.horizontal, 6)
            .padding(.vertical, 2)
            .background(Brand.proGold.opacity(0.18), in: RoundedRectangle(cornerRadius: 6))
            .accessibilityLabel(String(localized: "Offhand Pro"))
    }
}

File: core/designsystem/src/main/kotlin/com/dmytrosamoilov/offhand/core/designsystem/component/ProCrown.kt
package com.dmytrosamoilov.offhand.core.designsystem.component

import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.dmytrosamoilov.offhand.core.designsystem.R

// The gold crown that marks a Pro-only action for a free user; the same art
// ships as ProCrown.imageset on iOS. Gold is a brand mark, so it is not tinted
// by the theme.
@Composable
fun ProCrown(
    modifier: Modifier = Modifier,
    size: Dp = 18.dp,
) {
    Image(
        painter = painterResource(R.drawable.ic_pro_crown),
        contentDescription = stringResource(R.string.designsystem_pro_badge),
        modifier = modifier.size(size),
    )
}
